kitsune/questions/handlers.py

```diff
@@ -0,0 +1,95 @@
+from dataclasses import dataclass, field
+
+from django.conf import settings
+from django.contrib.auth.models import User
+
+from kitsune.questions.models import Answer, Question
+from kitsune.sumo.handlers import AbstractChain, AccountHandler
+from kitsune.users.handlers import UserDeletionListener
+
+
+class SpamAAQHandler(AccountHandler):
+    """Handler for spam cleanup."""
+
+    def handle_account(self, data: dict) -> None:
+        """Handler to delete all spam questions and answers for a given user."""
+        questions = data["questions"]
+        answers = data["answers"]
+
+        spam_questions = questions.filter(is_spam=True).values_list("id", flat=True)
+        spam_answers = answers.filter(is_spam=True).values_list("id", flat=True)
+        Question.objects.filter(id__in=spam_questions).delete()
+        Answer.objects.filter(id__in=spam_answers).delete()
+
+
+class ArchivedProductAAQHandler(AccountHandler):
+    """Handler for archived product cleanup."""
+
+    def handle_account(self, data: dict) -> None:
+        """Handler to delete all forum posts for a given user for archived products."""
+        questions = data["questions"]
+        answers = data["answers"]
+
+        archived_questions = questions.filter(product__is_archived=True).values_list(
+            "id", flat=True
+        )
+        archived_answers = answers.filter(question__in=archived_questions).values_list(
+            "id", flat=True
+        )
+        Question.objects.filter(id__in=archived_questions).delete()
+        Answer.objects.filter(id__in=archived_answers).delete()
```

Comment:
I think these two should be switched in order of execution due to the dependency of `archived_answers` on `archived_questions`? Otherwise, I think if you delete the questions first, the query for the `archived_answers` won't return anything so no answers will ever be deleted?

Oh, actually, when I noticed that you had tests that run successfully for this, I then realized that it works only because of the cascade-delete [here](https://github.com/mozilla/kitsune/blob/c10de5dd2360f47860d0552e3a0e5abdfec54002/kitsune/questions/models.py#L873). So I don't think you even need to delete the answers, because they'll be cascade-deleted when the questions are deleted.